lightning/src/util/ser.rs

```diff
@@ -677,6 +678,8 @@ macro_rules! impl_consensus_ser {
 }
 impl_consensus_ser!(Transaction);
 impl_consensus_ser!(TxOut);
+impl_consensus_ser!(Txid);
```

Comment:
Can we not do these directly by accessing the [u8; 32] and serializing based on that?

Reply:
With using consensus serialization it's more reliable since it accounts for byte order (as a part of `hash_newtype!` macro behavior)

Reply:
Hmm, I don't think we *want* to account for byte order when serializing txids/bitcoin block hashes to other lightning peers - it would not be strange to use a different byte order in Lightning vs Bitcoin P2P.

Reply:
Also, wait, I'm confused, the current implementation in rust-bitcoin dosn't do anything but write the underlying. https://github.com/rust-bitcoin/rust-bitcoin/blob/a2bfcb5a894ca2bf2931bfdd99372052b9755a01/src/hash_types.rs#L27

Reply:
Now I am confused :)

The macro simply uses the serialization of underlying sha256d value; I am not sure it worth re-writing it's serialization method to diverge from bitcoin's implementation...

Reply:
We already implement Writeable/Readable for Sha256dHash, as a function writing a 32-byte slice directly. Doing it this way here means we rely on creating a temporary wrapper, serializing through that, then mapping the return value back into our own types.

Reply:
Done